save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt

```diff
@@ -257,9 +274,13 @@ private suspend fun ComponentWithScope<*, *>.loadingHandler(request: suspend ()
 
 private fun ComponentWithScope<*, *>.withModalResponseHandler(
     response: Response,
+    isNeedRedirect: Boolean
 ) {
     if (!response.ok) {
         val statusContext: RequestStatusContext = this.asDynamic().context
+        if (isNeedRedirect) {
+            statusContext.setIsNeedRedirect(isNeedRedirect)
+        }
```

Comment:
```suggestion
    isNeedRedirect: Boolean = false,
) {
    if (!response.ok) {
        val statusContext: RequestStatusContext = this.asDynamic().context
        statusContext.setIsNeedRedirect(isNeedRedirect)
```
It seems that with the current approach after `responseHandlerWithRedirect` gets called once, all consecutive `404`s will cause redirect, because the flag hasn't been unset.

Reply:
Yes, agree. Fixed